package.xml

```diff
@@ -1,24 +1,29 @@
 <?xml version="1.0"?>
-<package>
+<?xml-model
+  href="http://download.ros.org/schema/package_format3.xsd"
+  schematypens="http://www.w3.org/2001/XMLSchema"?>
+<package format="3">
   <name>genpy</name>
   <version>0.6.8</version>
   <description>Python ROS message and service generators.</description>
   <maintainer email="dthomas@osrfoundation.org">Dirk Thomas</maintainer>
   <license>BSD</license>
 
+  <url type="website">http://wiki.ros.org/genpy</url>
   <url type="bugtracker">https://github.com/ros/genpy/issues</url>
   <url type="repository">https://github.com/ros/genpy</url>
 
   <author>Ken Conley</author>
   <author>Troy Straszheim</author>
   <author>Morten Kjaergaard</author>
+  <author>Dirk Thomas</author>
 
-  <buildtool_depend version_gte="0.5.78">catkin</buildtool_depend>
+  <depend>genmsg</depend>
 
-  <build_depend>genmsg</build_depend>
+  <buildtool_depend version_gte="0.5.78">catkin</buildtool_depend>
 
-  <run_depend>genmsg</run_depend>
-  <run_depend>python-yaml</run_depend>
+  <exec_depend condition="$ROS_PYTHON_VERSION == 2">python-yaml</exec_depend>
+  <exec_depend condition="$ROS_PYTHON_VERSION == 3">python3-yaml</exec_depend>
```

Comment:
Are we not able to just remap this using rosdep? If every package has to double up their python debian dependencies like this, that's going to be a nightmare.

Reply:
> Are we not able to just remap this using rosdep?

That kind of functionality is not planned.

> If every package has to double up their python debian dependencies like this, that's going to be a nightmare.

First this only applies to packages which want to support Melodic and Noetic from the same branch. If you use a separate branch you just change the existing entry from the `python-` key to the `python3-` key.

Also the overhead of duplicating these lines once doesn't strike me as a "nightmare" - it is more a small inconvenience.

Reply:
Not planned? It was already complete and working: https://github.com/ros/rosdistro/search?q=xenial_python3

Reply:
The approach of using entries with the `_python3` suffix was aimed for the use case that a single distro might be used with either Python 2 or Python 3. Since that is not the case with the hard switch to Python 3 in Noetic the use of separate keys avoids some drawbacks of the suffix approach.

Reply:
See ros-infrastructure/rep#201.

Reply:
Excellent, thanks for the reference— an interesting read.